server/history/statistics.py

```diff
@@ -0,0 +1,48 @@
+import csv
+import datetime
+from server.s3 import download
+
+BUCKET = "tm-mbta-performance"
+CACHE = {}
+CACHE_LAST_REFRESH_DAY = None
+ROUTES = ["Red", "Orange"]
+S3_KEY_NEW_TRAINS = "NewTrains/run_counts/{}.csv"
+
+
+def download_history(route):
+    key = S3_KEY_NEW_TRAINS.format(route)
+    return download(BUCKET, key, compressed=False)
+
+
+def get_cached_summaries():
+    global CACHE
+    global CACHE_LAST_REFRESH_DAY
+    now = datetime.datetime.utcnow()
+    todays_data_upload = now.replace(hour=10, minute=6, second=0, microsecond=0)
+    if CACHE_LAST_REFRESH_DAY is None or (CACHE_LAST_REFRESH_DAY != now.date() and now > todays_data_upload):
```

Comment:
first condition should maybe check the CACHE itself and not just the day, just in case? idk

Reply:
oh true, cannot hurt to be explicit about that.